src/combineReducers.js

```diff
@@ -26,7 +25,7 @@ function getUnexpectedStateShapeWarningMessage(inputState, reducers, action, une
     )
   }
 
-  if (!isPlainObject(inputState)) {
+  if (!inputState || inputState.constructor !== Object) {
```

Comment:
This seems to have the same problem with an old-fashioned `isArray` implementation. It always fails if the action comes from another window.